mindmeld/active_learning/data_loading.py

```diff
@@ -0,0 +1,425 @@
+            log_queries, _, _, _ = StrategicRandomSampling(sample_size).sample(
+                unsampled=log_queries
+            )
+        return DataBucket(
+            label_map=LabelMap.get_label_map(app_path, train_pattern),
+            test_queries=QueryLoader(app_path, test_pattern, load, save).queries,
+            unsampled_queries=log_queries,
+            sampled_queries=QueryLoader(app_path, train_pattern, load, save).queries,
+        )
+
+
+class DataBucket:
+    """Class to hold data throughout the Active Learning training pipeline.
+    Responsible for data conversion, filtration, and storage.
+    """
+
+    def __init__(
+        self,
+        label_map,
+        test_queries,
+        unsampled_queries,
+        sampled_queries,
+        newly_sampled_queries=None,
+    ):
+
+        self.label_map = label_map
+        self.test_queries = test_queries
+        self.unsampled_queries = unsampled_queries
+        self.sampled_queries = sampled_queries
+        self.newly_sampled_queries = (
+            newly_sampled_queries
+            if newly_sampled_queries
+            else deepcopy(sampled_queries)
+        )
+
+    def convert_queries_to_tuples(self, queries: List, class_type: str = "domain"):
+        """Converts MindMeld queries into queries in a tuple format.
+
+        Args:
+            queries (list): List of queries to convert
+            class_type (str): Queries can map to "domain" or "intent" labels
+
+        Returns:
+            tuple_queries (list): List of queries in a tuple format. [(Text, Class)...]
+        """
+        tuple_queries = []
+
+        if class_type == "domain":
+            for query in queries:
+                encoded_domain = self.label_map.domain2id[query.domain]
+                tuple_queries.append((query.query.text, encoded_domain))
+        elif class_type == "intent":
+            for query in queries:
+                encoded_intent = self.label_map.intent2id[query.domain][query.intent]
+                tuple_queries.append((query.query.text, encoded_intent))
+        else:
+            raise ValueError(
+                f"Class type {class_type} is invalid. Must be 'domain' or 'intent'."
+            )
+        return tuple_queries
+
+    @staticmethod
+    def filter_queries(queries: List, domain: str, intent: str = None):
+        """Filter queries for training preperation.
+
+        Args:
+            queries (list): List of queries to filter
+            domain (str): Domain of desired queries
+            intent (str): Intent of desired queries, can be None.
+
+        Returns:
+            filtered_queries_indices (list): List of indices of filtered queries.
+            filtered_queries (list): List of filtered queries.
+        """
+        filtered_queries_indices = []
+        filtered_queries = []
+        for index, query in enumerate(queries):
+            if query.domain == domain:
+                if not intent or (intent and query.intent == intent):
```

Comment:
why do we still include the query if it doesnt have an intent? An explaination here would be helpful.